modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java

```diff
@@ -462,4 +591,43 @@ public void handleReferenceHighlighting(ReferenceHighlightingObject references)
     public void removeReferenceHighlighting() {
         view.getGutterAnnotations().forEach(gutterAnnotation -> {gutterAnnotation.setProofNodeIsReferenced(false);});
     }
+
+    /**
+     * Listener to act upon a change of the insertion position
+     */
+    private class PositionChangedListener implements ChangeListener<Position> {
```

Comment:
Actually if you enable disable the listeners in my opinion it is more transparent what happens. otherwise you have to keep track using flags or Observables about the current state. i think this may become difficult to keep track of. @mattulbrich what is your opinion?